src/ctapipe/utils/astro.py

```diff
@@ -4,63 +4,231 @@
 
 __all__ = ["get_bright_stars"]
 
 
-def get_bright_stars(pointing=None, radius=None, magnitude_cut=None):
+CACHE_FILE = Path("~/.psf_stars.ecsv").expanduser()
+
+
+class StarCatalogues(Enum):
+    Yale = "V/50/catalog"  # Yale bright star catalogue
+    Hippoarcos = "I/239/hip_main"  # hipparcos catalogue
+
+
+def select_stars(stars, pointing=None, radius=None, Bmag_cut=None, Vmag_cut=None):
+    """
+    utility function to cut stars based on magnitude and/or location
+
+    Parameters
+    ----------
+    stars: astropy table
+        Table of stars, including magnitude and coordinates
+    pointing: astropy Skycoord
+        pointing direction in the sky (if none is given, full sky is returned)
+    radius: astropy angular units
+        Radius of the sky region around pointing position. Default: full sky
+    Vmag_cut: float
+        Return only stars above a given apparent magnitude. Default: None (all entries)
+    Bmag_cut: float
+        Return only stars above a given absolute magnitude. Default: None (all entries)
+
+    Returns
+    -------
+    Astropy table:
+       List of all stars after cuts with same keys as the input table stars
     """
-    Get an astropy table of bright stars.
 
-    This function is using the Yale bright star catalog, available through ctapipe
-    data downloads.
+    if Bmag_cut is not None and "Bmag" in stars.keys():
+        stars = stars[stars["Bmag"] < Bmag_cut]
+    elif Bmag_cut is not None and "BTmag" in stars.keys():
+        stars = stars[stars["BTmag"] < Bmag_cut]
+    if Vmag_cut is not None and "Vmag" in stars.keys():
+        stars = stars[stars["Vmag"] < Vmag_cut]
+    elif Vmag_cut is not None and "VTmag" in stars.keys():
+        stars = stars[stars["VTmag"] < Vmag_cut]
 
-    The included Yale bright star catalog contains all 9096 stars, excluding the
-    Nova objects present in the original catalog and is complete down to magnitude
-    ~6.5, while the faintest included star has mag=7.96. :cite:p:`bright-star-catalog`
+    if radius is not None:
+        if pointing is None:
+            raise ValueError(
+                "Sky pointing, pointing=SkyCoord(), must be "
+                "provided if radius is given."
+            )
+        separations = stars["ra_dec"].separation(pointing)
+        stars["separation"] = separations
+        stars = stars[separations < radius]
+
+    return stars
+
+
+def get_star_catalog(catalog, filename):
+    """
+    Utility function to download a star catalog for the get_bright_stars function
+
+    Parameters
+    ----------
+    catalog: string
+        Name of the catalog to be used. Usable names are found in the Enum StarCatalogues. Default: Yale
+    filename: string
+        Name and location of the catalog file to be produced
+    """
+    from astroquery.vizier import Vizier
+
+    vizier = Vizier(
+        catalog=StarCatalogues[catalog].value,
+        columns=[
+            "recno",
+            "RAJ2000",
```

Comment:
Please remove the computed "RAJ2000" / "DEJ2000" values. 

These are computed by vizier on the fly and are redundant with the original coordinates and the proper motions.

We anyway need to apply proper motion, so it doesn't matter whether we start from the catalog epoch or j2000

Reply:
For the Yale catalog there is no other good coordinates other than RAJ2000 and DEJ2000. I made it the same way as the old function now where the coorinates from the catalog are deleted and you just keep the SkyCoord objects for the star position, this time just with proper motion.

Reply:
Ok, then let's use the precomputed RAJ2000 consistently, which also makes it easier since we can use the same epoch for both catalogs. Remove the other RA/DEC then.

Reply:
I removed the other coordinate, but the functions for the two catalogues are almost identical now. I can have just one function and bring the catalog name as an argument.

Reply:
Yes, please factor out common code into a lower-level function.

I expect this to look basically like this:

```
def get_bright_stars(...):
    stars = get_table_dataset("yale_bright_stars.fits.gz")
    return _get_stars(stars, ...)
```
and the same for the other catalog